conan/cli/cli.py

```diff
@@ -45,6 +45,7 @@ def add_commands(self):
             Cli._builtin_commands = self._commands.copy()
         else:
             self._commands = Cli._builtin_commands.copy()
+            self._groups = defaultdict(list)
```

Comment:
This fixes the issue of a user directly instantiating ``Cli`` and using ``cli.run()`` themselves, in which groups are never cleaned and accumulate at each call.